artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQConnection.java

```diff
@@ -97,7 +98,7 @@
 
    private final Set<SimpleString> tempQueues = new ConcurrentHashSet<>();
 
-   private final Set<SimpleString> knownDestinations = new ConcurrentHashSet<>();
+   private final ConcurrentMaxSizeCache<SimpleString> knownDestinations = new ConcurrentMaxSizeCache<>(100);
```

Comment:
Can we make this configurable (e.g. a setting in the CF), 100 is a bit of a magic number.

Reply:
this will actually cause issues if someone is using temp destinations, and has say more than 100 legitimately. and creates the consumers later, as the consumer will check it its known, which now it wont and will throw an error. 

This will at least mean people can increase that setting if they need.